internal/groups/api/endpoint.go

```diff
@@ -0,0 +1,316 @@
+package groups
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/internal/groups"
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+func CreateGroupEndpoint(svc groups.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createGroupReq)
+		if err := req.validate(); err != nil {
+			return groupRes{}, err
+		}
+
+		group := groups.Group{
+			Name:        req.Name,
+			Description: req.Description,
+			ParentID:    req.ParentID,
+			Metadata:    req.Metadata,
+		}
+
+		g, err := svc.CreateGroup(ctx, req.token, group)
```

Comment:
It is unclear from the code (on reading) what is the difference between `group` and `g` - they both suggest to be groups. It is also expected for `CreateGroup()` to return group, but on the other hand you created it manually above.

Reply:
same as `CreateThings` works